lab-ron/app/component/app-gallery/app-gallery-controller.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const angular = require('angular');
+
+angular.module('demoApp')
+.controller('AppGalleryController', [
+  AppGalleryController
+]);
+
+function AppGalleryController(){
+  this.galleryItem = {
+    title: '',
+    desc: '',
+    imgSrc: ''
+  };
+
+  this.addGalleryItem = function(){
+    this.images.push(angular.copy(this.galleryItem));
+    this.galleryItem = {title: '', desc: '', imgSrc: ''};
+  };
+  this.images = [
```

Comment:
It would probably be more readable if you broke this out into its own file, then required it in.